p2p/peer.py

```diff
@@ -610,20 +610,20 @@ class ETHPeer(BasePeer):
             self.disconnect(DisconnectReason.other)
             raise HandshakeFailure(
                 "Expected a ETH Status msg, got {}, disconnecting".format(cmd))
-        msg = cast(Dict[str, Any], msg)
-        if msg['network_id'] != self.network_id:
+        msg = cast(Dict[bytes, Any], msg)
+        if msg[b'network_id'] != self.network_id:
             self.disconnect(DisconnectReason.other)
             raise HandshakeFailure(
                 "{} network ({}) does not match ours ({}), disconnecting".format(
-                    self, msg['network_id'], self.network_id))
+                    self, msg[b'network_id'], self.network_id))
         genesis = await self.genesis
-        if msg['genesis_hash'] != genesis.hash:
+        if msg[b'genesis_hash'] != genesis.hash:
```

Comment:
I'm slightly worried about using `bytes` as dict keys here since we're kinda wired to writing dict keys as string literals, and although mistakenly trying to access a dict item here using something other than `bytes` **could** be caught by mypy, it could lead to hard-to-debug issues when not caught by mypy

Reply:
I agree, I'm not satisfied with this solution.  I'll rework this to do the conversion between bytes/strings at the proper location.